packages/flutter/lib/src/services/text_input.dart

```diff
@@ -2532,11 +2532,18 @@ class SystemContextMenuController with SystemContextMenuClient {
   /// Null if [show] has not been called.
   Rect? _lastTargetRect;
 
+  /// The [IOSSystemContextMenuItem]s that were last given to [show].
+  ///
+  /// Null if [show] has not been called.
+  List<IOSSystemContextMenuItemData>? _lastItems;
+
   /// True when the instance most recently [show]n has been hidden by the
   /// system.
   bool _hiddenBySystem = false;
 
-  bool get _isVisible => this == _lastShown && !_hiddenBySystem;
+  /// Indicates whether the system context menu managed by this controller is
+  /// currently being displayed to the user.
+  bool get isVisible => this == _lastShown && !_hiddenBySystem;
```

Comment:
Was the main reason this was made public because of tests? If so would `@VisibleForTesting` be appropriate here?

Reply:
Yes but I started thinking it might be useful for someone that's managing their own system context menu... What do you think? I could always remove `@VisibleForTesting` later if someone asks for it I guess.

Reply:
I see, if this is similar to `ContextMenuController.isShown` then it sounds reasonable to have. I'll leave it up to you.